core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -841,18 +857,43 @@ private Function compileFilter(
         }
         if (!threadSafe) {
             ObjList<ObjList<GroupByFunction>> allWorkerGroupByFunctions = new ObjList<>();
-            for (int i = 0; i < workerCount; i++) {
-                ObjList<GroupByFunction> workerGroupByFunctions = new ObjList<>(groupByFunctions.size());
-                allWorkerGroupByFunctions.extendAndSet(i, workerGroupByFunctions);
-                GroupByUtils.prepareWorkerGroupByFunctions(
-                        model,
-                        metadata,
-                        functionParser,
-                        executionContext,
-                        groupByFunctions,
-                        workerGroupByFunctions
-                );
+            boolean supportDeepClone = true;
+            try {
+                for (int i = 0; i < workerCount; i++) {
```

Comment:
`GroupByUtils#prepareWorkerGroupByFunctions()` does more than function cloning, e.g. it calls `initValueIndex()` on the group by functions. So, it might be a good idea to move the `deepClone()` attempt into the `prepareWorkerGroupByFunctions()` method.